### PR TITLE
Feature/aa 1162 get account managers

### PR DESCRIPTION
**Pull request by DaniyarAskarov**
Issue: https://github.com/happy-travel/agent-app-project/issues/1162
Tested through insomnia

Test: 
1. Edo -> Admin -> Agencies (AgencyInfo and AdminViewAgencyInfo models were extended)
2. Edo -> Admin -> AccountManagers

AccountManagerId was added to Agency entity and for a few models.

### CODE REVIEW

File: Api/Extensions/AdministratorExtensions.cs [hunk 1]
```diff
@@ -13,5 +14,17 @@ public static AdministratorInfo ToAdministratorInfo(this Administrator administr
...
+        public static AccountManager ToAccountManager(this Administrator administrator)
+            => new AccountManager(
+                id: administrator.Id,
+                firstName: administrator.FirstName,
+                lastName: administrator.LastName,
+                position: administrator.Position);
+
+
+        public static string GetFullName(this Administrator administrator)
+            => $"{administrator.LastName} {administrator.FirstName}";
```
Comment (kirillta) [inline Api/Extensions/AdministratorExtensions.cs:28]: First name must go before last name. And I'm sure the method must be places somewhere [here](https://github.com/happy-travel/data-formatters/tree/master/HappyTravel.DataFormatters)

File: Api/Models/Agencies/AdminViewAgencyInfo.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ namespace HappyTravel.Edo.Api.Models.Agencies
         /// <summary>
         /// Name of the account manager
         /// </summary>
-        public string AccountManagerName { get; init; }
+        public string? AccountManagerName { get; init; }
```
Comment (kirillta) [inline Api/Models/Agencies/AdminViewAgencyInfo.cs:41]: Probably we must assign a default manager when an agency creates. @AlexeyShok what do you think?

File: HappyTravel.Edo.Data/Agents/Agency.cs [hunk 3]
```diff
@@ -33,5 +33,6 @@ public class Agency
         public string LegalAddress { get; set; } = string.Empty;
         public PaymentTypes PreferredPaymentMethod { get; set; }
         public bool IsContractUploaded { get; set; }
+        public int? AccountManagerId { get; set; }
```
Comment (kirillta) [inline HappyTravel.Edo.Data/Agents/Agency.cs:36]: Same as above

File: Api/Models/Agencies/AdminViewAgencyInfo.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Models/Agencies/AdminViewAgencyInfo.cs:41]: I don't think this is necessary. All the same, the agency must be verified by the admin. During verification, he can choose a manager account for the agency. For convenience, when registering a child agency, you can specify the account of the manager of the root agency in it.

File: Api/AdministratorServices/AdministratorManagementService.cs [hunk 4]
```diff
@@ -65,6 +73,8 @@ private async Task<Result<Administrator>> Get(int administratorId)
                 ?? Result.Failure<Administrator>("Administrator with specified Id does not exist");
 
 
+        private const int AccountManagerRole = 1;
```
Comment (AlexeyShok) [inline Api/AdministratorServices/AdministratorManagementService.cs:76]: AccountManagerRoleId

File: Api/Controllers/AdministratorControllers/AccountManagersController.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+
+        /// <summary>
+        ///     Gets all account managers
+        /// </summary>
+        /// <returns>List of account managers' info</returns>
+        [HttpGet("account-managers")]
+        [ProducesResponseType(typeof(List<AdministratorInfo>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int)HttpStatusCode.BadRequest)]
```
Comment (AlexeyShok) [inline Api/Controllers/AdministratorControllers/AccountManagersController.cs:31]: [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]

File: Api/Models/Management/Administrators/AccountManager.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+namespace Api.Models.Management.Administrators
+{
+    public class AccountManager
```
Comment (AlexeyShok) [inline Api/Models/Management/Administrators/AccountManager.cs:3]: Maybe struct or record?

File: Api/Models/Agencies/AdminViewAgencyInfo.cs (hunk 2 above)
Comment (kirillta) [reply Api/Models/Agencies/AdminViewAgencyInfo.cs:41]: I agree, but if we implement default admins, we reduce a chance when an agency gets lost after the registration

**Review by kirillta — APPROVED** [review]

File: Api/AdministratorServices/AdminAgencyManagementService.cs [hunk 7]
```diff
@@ -49,21 +50,27 @@ public Task<Result> ActivateAgency(int agencyId, string reason)
...
+                            ? rootAgency.Verified.Value.DateTime
                             : null,
-                        c.Names,
+                        country.Names,
                         languageCode,
                         markupFormula == null
                             ? string.Empty
-                            : markupFormula.DisplayFormula))
+                            : markupFormula.DisplayFormula,
+                        PersonNameFormatters.ToMaskedName(admin.FirstName, admin.LastName, null)))
```
Comment (kirillta) [inline Api/AdministratorServices/AdminAgencyManagementService.cs:73]: Not sure if we need exactly the masked formatter

**Review by AlexeyShok — APPROVED** [review]

**Review by kirillta — APPROVED** [review]

**Pull request merged**